### PR TITLE
Logs and events on API

### PR DESCRIPTION
**Pull request by miiu96**
Extend the API route `/transaction/:hash?withResults=true`
Nowthis API will return also the logs and events generated by a transaction or by smart contract results.

### CODE REVIEW

File: node/nodeTransactionsResults_test.go [hunk 1]
```diff
@@ -174,14 +181,101 @@ func TestPutEventsInTransactionSmartContractResults(t *testing.T) {
...
+			{
+				Address:    []byte("addr2"),
+				Identifier: []byte(core.BuiltInFunctionESDTBurn),
+				Topics:     [][]byte{[]byte("topic1"), []byte("topic2")},
+				Data:       []byte("data1"),
+			},
+		},
+	}
+
+	marshalizerdMock := &mock.MarshalizerFake{}
```
Comment (bogdan-rosianu) [inline node/nodeTransactionsResults_test.go:222]: marshalizer

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: data/transaction/apiTransactionResult.go [hunk 2]
```diff
@@ -42,6 +42,7 @@ type ApiTransactionResult struct {
 	MiniBlockHash                     string                    `json:"miniblockHash,omitempty"`
 	Receipt                           *ReceiptApi               `json:"receipt,omitempty"`
 	SmartContractResults              []*ApiSmartContractResult `json:"smartContractResults,omitempty"`
+	Logs                              *LogsAPI                  `json:"logs,omitempty"`
```
Comment (bogdan-rosianu) [inline data/transaction/apiTransactionResult.go:45]: We have `ReceiptApi`, `ApiSmartContractResult` and `LogsAPI`. can we use a single pattern?

Comment (bogdan-rosianu) [inline data/transaction/apiTransactionResult.go:45]: We have `ReceiptApi`, `ApiSmartContractResult` and `LogsAPI`. can we use a single pattern?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]
as stated above :point_up:

File: process/smartContract/builtInFunctions/esdtNFTTransfer.go [hunk 3]
```diff
@@ -231,6 +231,11 @@ func (e *esdtNFTTransfer) processNFTTransferOnSenderShard(
 		return nil, err
 	}
 
+	tokenNonce := esdtData.TokenMetaData.Nonce
```
Comment (sasurobert) [inline process/smartContract/builtInFunctions/esdtNFTTransfer.go:234]: all these changes has to be put inside vm-common. not here. as this is getting deleted.

Comment (miiu96) [reply process/smartContract/builtInFunctions/esdtNFTTransfer.go:234]: done.

File: node/nodeTransactionsResults_test.go (hunk 1 above)
Comment (miiu96) [reply node/nodeTransactionsResults_test.go:222]: done.

File: data/transaction/apiTransactionResult.go (hunk 2 above)
Comment (miiu96) [reply data/transaction/apiTransactionResult.go]: done.

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**